spec/features/add_an_ip_address_spec.rb

```diff
@@ -9,61 +9,64 @@
   include_examples 'confirmation use case spy'
   include_examples 'notifications service'
 
-  context 'when logged out' do
+  context 'when logged in' do
     before do
+      sign_in_user create(:user)
       visit new_ip_path
     end
 
-    it_behaves_like 'not signed in'
-  end
+    it_behaves_like 'shows activation notice'
 
-  context 'when logged in' do
-    let(:user) { create(:user, :confirmed, :with_organisation) }
-    before do
-      sign_in_user user
-      visit ips_path
-      expect(page).to have_content "Add IP"
-      click_on "Add IP Address"
+    it 'asks me to enter an IP' do
+      expect(page).to have_content('Enter IP Address (IPv4 only)')
     end
 
-    context 'before saving the IP' do
-      it_behaves_like 'shows activation notice'
+    context 'and that IP is valid' do
+      before do
+        fill_in 'address', with: '10.0.0.1'
+        click_on 'Save'
+      end
+
+      it 'shows a success message' do
+        expect(page).to have_content('IP Added')
+      end
 
-      it 'displays the form' do
-        expect(page).to have_content('Enter IP Address (IPv4 only)')
+      it 'shows me a list of IPs including the new IP' do
+        expect(page).to have_content('10.0.0.1')
```

Comment:
I usually also assert on the current path, could get false positives as the show page would contain this as well

Reply:
Hmm, I actually don't like to do this - I think the path is an implementation detail, and what a user is really interested in is checking the content on the page.

Checking paths has the same problem as any other internals - can end up brittle, or fooling you into thinking something is safe under test when the _actual_ functionality is missing.

Reply:
If you want to check they're really seeing all the IPs, I happily could add another test for that.

Reply:
This was simple, so added a new spec [in this commit](https://github.com/alphagov/govwifi-admin/pull/83/commits/912474e69814186219d6755c8b7537dcf5391706).